src/tdastro/sources/redback_models.py

```diff
@@ -0,0 +1,200 @@
+            self.source = redback.model_library.all_models_dict[source]
+        else:
+            self.source_name = source.__name__
+            self.source = source
+
+        # Redback models already handle redshift, so we do not want to double apply it.
+        self.apply_redshift = False
+
+    @property
+    def param_names(self):
+        """Return a list of the model's parameter names."""
+        return self.source_param_names
+
+    def minwave(self, graph_state=None):
+        """Get the minimum wavelength of the model.
+
+        Parameters
+        ----------
+        graph_state : GraphState, optional
+            An object mapping graph parameters to their values. Not used
+            for this model.
+
+        Returns
+        -------
+        minwave : float or None
+            The minimum wavelength of the model (in angstroms) or None
+            if the model does not have a defined minimum wavelength.
+        """
+        return 0.0
+
+    def maxwave(self, graph_state=None):
+        """Get the maximum wavelength of the model.
+
+        Parameters
+        ----------
+        graph_state : GraphState, optional
+            An object mapping graph parameters to their values. Not used
+            for this model.
+
+        Returns
+        -------
+        maxwave : float or None
+            The maximum wavelength of the model (in angstroms) or None
+            if the model does not have a defined maximum wavelength.
+        """
+        return math.inf
+
+    def compute_flux(self, times, wavelengths, graph_state=None, **kwargs):
+        """Draw effect-free observations for this object.
+
+        Parameters
+        ----------
+        times : numpy.ndarray
+            A length T array of rest frame timestamps.
+        wavelengths : numpy.ndarray, optional
+            A length N array of wavelengths (in angstroms).
+        graph_state : GraphState
+            An object mapping graph parameters to their values.
+        **kwargs : dict, optional
+           Any additional keyword arguments.
+
+        Returns
+        -------
+        flux_density : numpy.ndarray
+            A length T x N matrix of SED values (in nJy).
+        """
+        params = self.get_local_params(graph_state)
+
+        # Build the function arguments from the parameter values.
+        fn_args = {}
+        for name in self.source_param_names:
+            fn_args[name] = params[name]
+
+        # Compute the shifted times.
+        t0 = params.get("t0", 0.0)
+        if t0 is None:
+            t0 = 0.0
+        shifted_times = times - t0
+
+        # Redback takes the wavelengths in nanometers and produces spectra in
+        # erg / cm^2 / s / Angstrom.  The spectra is oversampled in time, so we
+        # will need to interpolate it at the times of interest.
+        wavelengths_nm = wavelengths / 10.0
+        rb_result = self.source(
+            shifted_times,
+            lambda_array=wavelengths_nm,
```

Comment:
It actually looks like `redback` uses angstroms
https://github.com/nikhil-sarin/redback/blob/33a4a493a78dca8d0cdacdd38f38b2519884e38d/redback/transient_models/kilonova_models.py#L1424
https://github.com/nikhil-sarin/redback/blob/33a4a493a78dca8d0cdacdd38f38b2519884e38d/redback/utils.py#L255